doc/tutorials/waveforms.rst

```diff
@@ -0,0 +1,270 @@
+    (Ch 1, Sample 2), (Ch 2, Sample 2), ..., (Ch 12, Sample 2),
+    ...,
+    (Ch 1, Sample 10,000), (Ch 2, Sample 10,000), ..., (Ch 12, Sample 10,000)
+
+To decode the raw multiplex waveform data to a numpy :class:`~numpy.ndarray`
+you can use the :func:`~pydicom.waveforms.numpy_handler.multiplex_array`
+function. The following decodes and returns the raw data from the multiplex at
+*index* ``0`` within the *Waveform Sequence*:
+
+.. code-block:: python
+
+    >>> from pydicom.waveforms import multiplex_array
+    >>> raw = multiplex_array(ds, 0, as_raw=True)
+    >>> raw[0, 0]
+    80
+
+
+If (003A,0210) *Channel Sensitivity* is present within the multiplex's *Channel
+Definition Sequence* then the raw sample data needs to be corrected before it's
+in the quantity it represents. This correction is given by (sample + *Channel
+Baseline*) x *Channel Sensitivity* x *Channel Sensitivity Correction Factor*
+and will be applied when `as_raw` is ``False`` or when using the
+:meth:`Dataset.waveform_array()<pydicom.dataset.Dataset.waveform_array>`
+function:
+
+    >>> arr = ds.waveform_array(0)
+    >>> arr[0, 0]
+    >>> 100.0
+    >>> import matplotlib.pyplot as plt
+    >>> fig, (ax1, ax2) = plt.subplots(2)
+    >>> ax1.plot(raw[:, 0])
+    >>> ax1.set_ylabel("unitless")
+    >>> ax2.plot(arr[:, 0])
+    >>> ax2.set_ylabel("μV")
+    >>> plt.show()
+
+.. image:: waveforms_assets/waveforms_decode.png
+   :width: 800
+   :align: center
+
+When processing large amounts of waveform data it might be more efficient to
+use the :func:`~pydicom.waveforms.numpy_handler.generate_multiplex` function
+instead. It yields an :class:`~numpy.ndarray` for each multiplex group
+within the *Waveform Sequence*:
+
+.. code-block:: python
+
+    >>> from pydicom.waveforms import generate_multiplex
+    >>> for arr in generate_multiplex(ds, as_raw=False):
+    ...     print(arr.shape)
+    ...
+    (10000, 12)
+    (1200, 12)
+
+
+Encoding *Waveform Data*
+========================
+
+Having seen how to decode and view a waveform then next step is creating our
+own multiplex group. The new group will contain two channels
+representing cosine and sine curves. We've chosen to represent our waveforms
+using signed 16-bit integers, but you can use signed or unsigned 8, 16, 32 or
+64-bit integers depending on the requirements of the IOD.
+
+First we create two :class:`ndarrays<numpy.ndarray>` with our waveform data:
+
+.. code-block:: python
+
+    >>> import numpy as np
+    >>> x = np.arange(0, 4 * np.pi, 0.1)
+    >>> ch1 = (np.cos(x) * (2**15 - 1)).astype('int16')
+    >>> ch2 = (np.sin(x) * (2**15 - 1)).astype('int16')
+
+Next we create the new multiplex group that will contain the waveforms:
+
+.. code-block:: python
+
+    >>> from pydicom.dataset import Dataset
+    >>> new = Dataset()
+    >>> new.WaveformOriginality = "ORIGINAL"
+    >>> new.NumberOfWaveformChannels = 2
+    >>> new.NumberOfWaveformSamples = len(x)
+    >>> new.SamplingFrequency = "1000"
```

Comment:
Why sampling frequency a string?

Reply:
VR is **DS**, can be either a string or a float. I chose string because that's how it printed earlier.

Reply:
I might change it at some point, I can understand how it might confuse people